File: backend/app/models/goal.py
import uuid
from datetime import datetime, timezone
from enum import Enum as PyEnum
from sqlalchemy import Column, String, Boolean, DateTime, ForeignKey, Enum
from sqlalchemy.orm import relationship
from app.database import Base


class GoalColor(str, PyEnum):
    blue = "blue"
    green = "green"
    amber = "amber"
    rose = "rose"
    violet = "violet"
    cyan = "cyan"


class Goal(Base):
    __tablename__ = "goals"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String(36), ForeignKey("users.id"), nullable=False)
    name = Column(String(100), nullable=False)
    color = Column(Enum(GoalColor), default=GoalColor.blue)
    archived = Column(Boolean, default=False)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

    user = relationship("User", back_populates="goals")
    tasks = relationship("Task", back_populates="goal", cascade="all, delete-orphan")

File: frontend/src/app/core/models/goal.ts
export type GoalColor = 'blue' | 'green' | 'amber' | 'rose' | 'violet' | 'cyan';

export const GOAL_COLORS: Record<GoalColor, string> = {
  blue: '#3B82F6',
  green: '#22C55E',
  amber: '#F59E0B',
  rose: '#F43F5E',
  violet: '#8B5CF6',
  cyan: '#06B6D4',
};

export interface Goal {
  id: string;
  user_id: string;
  name: string;
  color: GoalColor;
  archived: boolean;
  created_at: string;
}

export interface GoalCreate {
  name: string;
  color: GoalColor;
}

export interface GoalUpdate {
  name?: string;
  color?: GoalColor;
  archived?: boolean;
}
